[OSDOCS#18862]: Added conditions for TNF in backup and restore (#109960)
openshift/openshift-docs: modules/dr-restoring-cluster-state.adoc

:_mod-docs-content-type: PROCEDURE
[id="dr-scenario-2-restoring-cluster-state_{context}"]
= Restoring to an earlier cluster state for more than one node

[role="_abstract"]
You can use a saved etcd backup to restore an earlier cluster state or restore a cluster that has lost the majority of control plane hosts.

> For a 2-node cluster with fencing (TNF) setup, a single surviving node can continue to operate in degraded mode. Use a saved etcd backup to restore an earlier cluster state if only one node is operational, or when both nodes have failed and you need to restart the cluster from a known safe state. In both cases, perform the restore procedure on a single node. The peer node automatically synchronizes its data with the restored node when it rejoins the cluster.

Reviewer: ```suggestion
For a Two-Node with Fencing (TNF) setup, a single surviving node can continue to operate in degraded mode. Use a saved etcd backup to restore an earlier cluster state if only one node is operational, or when both nodes have failed and you need to restart the cluster from a known safe state. In both cases, perform the restore procedure on a single node. The peer node automatically synchronizes its data with the restored node when it rejoins the cluster.
```

> For high availability (HA) clusters, to maintain high availability and prevent a cluster split, shut down etcd on the following number of hosts:

Reviewer: You changed the paragraph above to two-node from HA, yet here we have mention of HA again. Is this correct? Are we now talking about an HA two-node situation? Please verify

. Exit the SSH session.

. When the API responds, to turn off the etcd Operator quorum guard, run the following command:
+
[IMPORTANT]
====

> For a 2-node cluster with fencing (TNF) setup, do not:

Reviewer: ```suggestion
For a Two-Node with Fencing (TNF) setup, do not:
```

  * Change the etcd Operator quorum setting. 

  * Turn the etcd Operator quorum off.

  * Turn the etcd Operator quorum on back.
====
+
[source,terminal]
----
$ oc patch etcd/cluster --type=merge -p '{"spec": {"unsupportedConfigOverrides": {"useUnsupportedUnsafeNonHANonProductionUnstableEtcd": true}}}'
----

> . To monitor the recovery progress of the control plane, run the following command:

Reviewer: ```suggestion
. Monitor the recovery progress of the control plane by running the following command:
```

> It can take up to 15 minutes for the control plane to recover.

Reviewer: ```suggestion
It can take up to 15 minutes for the control plane to recover. Wait for the control plane to recover before using the next step.
```

====

> .  When recovered, to enable the quorum guard, run the following command:

Reviewer: ```suggestion
.  Enable the quorum guard by running the following command:
```

.Troubleshooting

> If you see no progress rolling out the etcd static pods, to manually force an etcd redeployment from the `cluster-etcd-operator`, run the following command:

Reviewer: ```suggestion
If the etcd static pods do not roll out , you can manually force an etcd redeployment from the `cluster-etcd-operator` by running the following command:
```
